pkgs/development/libraries/tbb/2020_3.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, gcc9Stdenv
+, fetchurl
+, fetchFromGitHub
+, fixDarwinDylibNames
+}:
+
+let
+  stdenv = gcc9Stdenv;
```

Comment:
`boost` is a c++ library that encodes compiler version into it's function names to make sure ABI does not rebak yb accident. I think it is crucial to use default compiler here (and frankly in every frequently used library C++ library in `nixpkgs`).

I would expect numerous build breakages when attempting to use it with a compiler different the one boost was built. Things will probably break even more if you build it on clang-based system.

Why do you need `gcc-9` here?

Reply:
The build fails with default gcc. See https://github.com/NixOS/nixpkgs/pull/215689#issuecomment-1435060127

GCC 9 is the latest supported version, so i thought we should use that.
Not sure how to handle clang here.

I don't know how this is usually handled in nixpkgs. It's the first time i have to deal with this.

Reply:
We had a patch that fixed the build for 2020.3. It works fine and is accepted by the upstream. Why do you want to change that?

Reply:
> The default compiler version is not supported.

The way I understand "not supported" is when major work is required  to adapt the software to the environment. It's a very rare occasion when old `gcc` just works and new `gcc` requires major work to adapt. `gcc` usually provides flags to get behaviour of an old compiler even on new `gcc` versions (be it disabling warnings, forcing an older version of C standard and whatnot). Mixing compiler versions and getting unique bugs is a real possibilty to get undebuggable problems.

Looking at https://github.com/oneapi-src/oneTBB/pull/866 the fix is trivial to backport. I would say it does not warrant using an outdated compiler. If that is the only problem I can help tweaking the build to succeed on modern `gcc`.

> Not sure how to handle clang here.

Usually it requires a more complex condition, like `if stdenv.isClang then stdenv else gcc9Stdenv`. But once you in that territory it's frequently easier to just fix the package.

Reply:
@chuangzhu said "GCC 12 isn't supported in v2021.8.0" so the obvious solution for me was to use the supported compiler, for both packages.

If it's consensus we can ignore upstream support status and better use the default version, we can try to get it working. I appreciate your help! I'm usually not dealing with such low level problems.

@trofi thanks for the explanation!